staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -0,0 +1,211 @@
+	return nil
+}
+
+func (t fakeApfFilter) Install(c *mux.PathRecorderMux) {
+}
+
+func newApfServer(decision int) *httptest.Server {
+	requestInfoFactory := &apirequest.RequestInfoFactory{APIPrefixes: sets.NewString("apis", "api"), GrouplessAPIPrefixes: sets.NewString("api")}
+	longRunningRequestCheck := BasicLongRunningRequestCheck(sets.NewString("watch"), sets.NewString("proxy"))
+
+	apfHandler := WithPriorityAndFairness(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	}), longRunningRequestCheck, fakeApfFilter{mockDecision: decision})
+	handler := apifilters.WithRequestInfo(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		r = r.WithContext(apirequest.WithUser(r.Context(), &user.DefaultInfo{
+			Groups: []string{user.AllUnauthenticated},
+		}))
+		apfHandler.ServeHTTP(w, r)
+	}), requestInfoFactory)
+
+	apfServer := httptest.NewServer(handler)
+	return apfServer
+}
+
+func TestApfSkipLongRunningRequest(t *testing.T) {
+	epmetrics.Register()
+
+	server := newApfServer(decisionError)
+	defer server.Close()
+
+	if err := expectHTTPGet(fmt.Sprintf("%s/api/v1/namespaces?watch=true", server.URL), http.StatusOK); err != nil {
+		// request should not be rejected
+		t.Error(err)
+	}
+
+	checkForExpectedMetrics(t, []string{
+		"apiserver_current_inflight_requests",
+	})
+}
+
+func TestApfRejectRequest(t *testing.T) {
+	epmetrics.Register()
+
+	server := newApfServer(decisionReject)
+	defer server.Close()
+
+	if err := expectHTTPGet(fmt.Sprintf("%s/api/v1/namespaces/default", server.URL), http.StatusTooManyRequests); err != nil {
+		t.Error(err)
+	}
+
+	// wait metrics to be collected
+	time.Sleep(time.Millisecond * 1500)
+
+	checkForExpectedMetrics(t, []string{
+		"apiserver_request_terminations_total",
+		"apiserver_dropped_requests_total",
+	})
+}
+
+func TestApfExemptRequest(t *testing.T) {
+	epmetrics.Register()
+	fcmetrics.Register()
+
+	// wait ReadWriteConcurrencyObserverPairGenerator to be registered
+	time.Sleep(time.Millisecond * 50)
+
+	server := newApfServer(decisionExempt)
+	defer server.Close()
+
+	if err := expectHTTPGet(fmt.Sprintf("%s/api/v1/namespaces/default", server.URL), http.StatusOK); err != nil {
+		t.Error(err)
+	}
+
+	// wait metrics to be collected
+	time.Sleep(time.Millisecond * 1500)
+
+	checkForExpectedMetrics(t, []string{
+		"apiserver_current_inflight_requests",
+		"apiserver_flowcontrol_read_vs_write_request_count_watermarks",
+		"apiserver_flowcontrol_read_vs_write_request_count_samples",
+	})
+}
+
+func TestApfExecuteRequest(t *testing.T) {
+	epmetrics.Register()
+	fcmetrics.Register()
+
+	// wait ReadWriteConcurrencyObserverPairGenerator to be registered
+	time.Sleep(time.Millisecond * 50)
+
+	server := newApfServer(decisionExecute)
+	defer server.Close()
+
+	if err := expectHTTPGet(fmt.Sprintf("%s/api/v1/namespaces/default", server.URL), http.StatusOK); err != nil {
+		t.Error(err)
+	}
+
+	// wait metrics to be collected
+	time.Sleep(time.Millisecond * 1500)
+
```

Comment:
This one also fails when I run it in commit 3121fe16a20e046d85419a4aff06edbe220b39b2:
```
--- FAIL: TestApfExecuteRequest (1.57s)
    priority-and-fairness_test.go:208: Scraped metrics did not include expected metric apiserver_current_inqueue_requests
```